hedera-node/src/main/java/com/hedera/services/sigs/verification/PrecheckKeyReqs.java

```diff
@@ -86,7 +87,7 @@ public PrecheckKeyReqs(
 			addQueryPaymentKeys(txn, keys);
 		}
 
-		return keys;
+		return keys.stream().distinct().collect(Collectors.toList());
```

Comment:
This may not work for `JKey`s that do not provide `equals()`, f.e. `JEd25519Key`

Reply:
We use Object.equals by default which works. Wrote a unit test here hedera-node/src/test/java/com/hedera/services/sigs/verification/PrecheckKeyReqsTest.java -> `usesBothOrderForQueryPayments()`